### PR TITLE
Adding Haydeez

### PR DESCRIPTION
**Pull request by GyrPhlymBaBumble**
## About The Pull Request
These changes add 8 new cyborg frames for 6 jobs and for syndicates. This involved updating robot_defines, uploading the files, and adding the robots to robot_model.dm

This was tested on a private server, and it all seemed to work properly. the borgs do not have resting animations, but do have open cover animations.

[Sprite](https://github.com/SPLURT-Station/S.P.L.U.R.T-Station-13/pull/919)

## Why It's Good For The Game
This PR adds attractive borgs that multiple users from the server have commented on positively about. While some may disparage the borg skins for being too horny, it does not seem unrealistic to me that someone would design a sexualized chassis for a being with no rights- in fact, it seems inevitable. These are just reskins, so they do not add any new mechanics outside of the visual

## Proof Of Testing
image
image
image

## Changelog

:cl: Flvffi
added: 8 Haydee borg skins for Engineering, Medical, Mining, Janitorial, Service, Peacekeeper, and Syndicate roles.
/:cl:

### CODE REVIEW

**Comment by projectkepler-ru** [comment]
Where are these from?

**Review by projectkepler-ru — CHANGES_REQUESTED** [review]
They'll need a resting animation, even if it's extremely crude, borg without resting animation will just become invisible

**Comment by theselfish** [comment]
> They'll need a resting animation, even if it's extremely crude, borg without resting animation will just become invisible

I'm pretty sure regular borgs don't have resting ams. 


And @GyrPhlymBaBumble . Best advise for testing is go to the ghost cafe and use the chameleon projector.

**Comment by ReturnToZender** [comment]
Haydeez nuts... gotem...

[This comment was marked as abuse]

**Comment by NargaInDisguise** [comment]
Stop using discord-link screenshots for PRs pleaese, they aren't hosted permanently anymore

**Comment by Boviro** [comment]
If I recall correctly, Bubberstation's intent was an RP server with ERP, rather than the other way around. I and many others just play the server to RP, rather than to ERP. Giving players a free pass to use very sexualized sprites in the halls feels like a step away from "an RP server with ERP elements." And to be clear I'm not saying ERP is bad, just that it's bleeding into the regular RP space more and more. If we want to stay an RP server and not an ERP server, we shouldn't really be adding goonerbait-tier tits and thighs as a borg chassis. I know you didn't make the sprites, but I would support this PR if the sexualization was toned down to like. Normal levels. Thanks for coming to my TED Talk chatroom

**Comment by theselfish** [comment]
> If I recall correctly, Bubberstation's intent was an RP server with ERP, rather than the other way around. I and many others just play the server to RP, rather than to ERP. Giving players a free pass to use very sexualized sprites in the halls feels like a step away from "an RP server with ERP elements." And to be clear I'm not saying ERP is bad, just that it's bleeding into the regular RP space more and more. If we want to stay an RP server and not an ERP server, we shouldn't really be adding goonerbait-tier tits and thighs as a borg chassis. I know you didn't make the sprites, but I would support this PR if the sexualization was toned down to like. Normal levels. Thanks for coming to my TED Talk chatroom

Okay so we're removing the lustwish, the robot dogs with giant asses and genitalia options right?

Thought not.

**Comment by NargaInDisguise** [comment]
> > If I recall correctly, Bubberstation's intent was an RP server with ERP, rather than the other way around. I and many others just play the server to RP, rather than to ERP. Giving players a free pass to use very sexualized sprites in the halls feels like a step away from "an RP server with ERP elements." And to be clear I'm not saying ERP is bad, just that it's bleeding into the regular RP space more and more. If we want to stay an RP server and not an ERP server, we shouldn't really be adding goonerbait-tier tits and thighs as a borg chassis. I know you didn't make the sprites, but I would support this PR if the sexualization was toned down to like. Normal levels. Thanks for coming to my TED Talk chatroom
> 
> Okay so we're removing the lustwish, the robot dogs with giant asses and genitalia options right?
> 
> Thought not.

We don't have dogborgs with giant asses, all we've got is raptor borgs that people also still think are a bit too goofy, me included.
Genitalia are hidded by several layers, clothing and toggles, and lustwishes are out of the way and something you have to directly interact with.

Point is, public indecency is a crime in space law, and this borg is definitely treading that line pretty hard, and that your reply misses the point entirely. I don't like this model, make it part of the dominatrix module or something please.

**Comment by NargaInDisguise** [comment]
Also these are called "boobrobot" in the code, theres really no beating around the bush that these are being added for anything other than the sex appeal, huh.

**Comment by NargaInDisguise** [comment]
> > They'll need a resting animation, even if it's extremely crude, borg without resting animation will just become invisible
> 
> I'm pretty sure regular borgs don't have resting ams.
> 
> And @GyrPhlymBaBumble . Best advise for testing is go to the ghost cafe and use the chameleon projector.

Also yeah, if you need to test the syndicate borg variant, just go sign up as a ghost cafe borg, the chameleon projector can select any variant, including ninja and syndie.

**Comment by Boviro** [comment]
> > If I recall correctly, Bubberstation's intent was an RP server with ERP, rather than the other way around. I and many others just play the server to RP, rather than to ERP. Giving players a free pass to use very sexualized sprites in the halls feels like a step away from "an RP server with ERP elements." And to be clear I'm not saying ERP is bad, just that it's bleeding into the regular RP space more and more. If we want to stay an RP server and not an ERP server, we shouldn't really be adding goonerbait-tier tits and thighs as a borg chassis. I know you didn't make the sprites, but I would support this PR if the sexualization was toned down to like. Normal levels. Thanks for coming to my TED Talk chatroom
> 
> Okay so we're removing the lustwish, the robot dogs with giant asses and genitalia options right?
> 
> Thought not.

The LW is a vending machine, there's not public sexualization to that, it's only sexual because you can get items that are generally supposed to be used only in dorms from it.
I wouldn't say raptorborgs are explictly sprited that way to be sexually appealing, as opposed to the haydee borgs this PR looks to add.
Exposed genitalia is punishable IC and on occasion OOC, it's a completely different ball game from having borgs explicitly added for sex appeal with no restrictions.
If you can somehow make this restricted to private areas (Like via the Dom module as Narga suggested), I would have no issues with adding it, but just adding it as a normal chassis gives people license to use it everywhere instead of keeping it to areas where everyone around has consented to seeing it.

**Comment by GyrPhlymBaBumble** [comment]
> Where are these from?



> 1. Do not discuss those details on the git if you have an issue bring it up on discord and have it be looked over by maintainer I couldn't careless we're adding more fetish to the game as much as it doesn't break anything and isn't content that'll get the server taken down. @Boviro  @NargaInDisguise @theselfish
> 
>     2. Where are these sprites from exactly?

The medical sprite is from splurt. I made some edits to it for them as it had stray and or missing pixels in 2023. I can find the change log for you once I get home. I made all of the others myself, originally for splurt station. 

> Also these are called "boobrobot" in the code, theres really no beating around the bush that these are being added for anything other than the sex appeal, huh.
> 
> No tact and a bit tasteless, imo

They are called boobrobot because that is what Boviro called them and fit the large and wide robot naming schemes. I was originally going to call them Haydee 



> They'll need a resting animation, even if it's extremely crude, borg without resting animation will just become invisible

I can test this but I don't think this is the case based on the other Borg sprites in the base code. I do plsn to eventually sprite wrecks and resting down the line.

**Comment by GyrPhlymBaBumble** [comment]
> > They'll need a resting animation, even if it's extremely crude, borg without resting animation will just become invisible
> 
> I'm pretty sure regular borgs don't have resting ams.
> 
> And @GyrPhlymBaBumble . Best advise for testing is go to the ghost cafe and use the chameleon projector.

https://discord.com/channels/1059199070016655462/1070767604559511552/1265400962818773103

Testing complete. it works. attempting to rest pulls up the message 'You can't do that!' will look at other borgs and see how they work

**Comment by GyrPhlymBaBumble** [comment]
> They'll need a resting animation, even if it's extremely crude, borg without resting animation will just become invisible

Untrue. They will be unable to rest. 
[CAN BORGS REST.txt](https://github.com/user-attachments/files/16354071/CAN.BORGS.REST.txt)

**Comment by StrangeWeirdKitten** [comment]
Right, going to lock the convo for now and look over the borg sprites and their code. Github isn't really the place for this. 

1. Changelog should be player facing communication of changes. Just say you added a borg sprite instead of putting down file paths.
2. Please use snippet tool to directly upload pictures with a copy and paste on github. The discord links are fine, but they're just going straight towards discord convos. 
3. Could you take a single picture with all the borg models? I don't need every facing direction, but it's good to layout and communicate the overall design layout for each department.

**Comment by StrangeWeirdKitten** [comment]
An example of point 3. 
image

**Comment by Boviro** [comment]
If I remember correctly discord image hosting is now time-limited so for preservation you may want to use github's embedding functions or another site like imgur. Just wanted to clear that up.

**Comment by GyrPhlymBaBumble** [comment]
> Right, going to lock the convo for now and look over the borg sprites and their code. Github isn't really the place for this.
> 
>     1. Changelog should be player facing communication of changes. Just say you added a borg sprite instead of putting down file paths.
> 
>     2. Please just use the inbuilt snippet tool to directly upload pictures with a copy and paste on github. The discord links are fine, but they're just going straight towards discord convos.
> 
>     3. Could you take a single picture with all the borg models? I don't need every facing direction, but it's good to layout and communicate the overall design layout for each department. (this is more for people who are later looking at this PR without having to delve into sprite diffs and dmi's)

image
from left to right: Row 1; Engineering, Medical, Mining, Janitor, Row 2; Service, Peacekeeper, Syndicate, Service (bunny ears, multiple users requested). I plan to eventually add a more detailed alt for most jobs, focusing on engineering next. I do not plan to add a cargo borg.

Syndicate was originally used for Security when I made it, but I was informed that bubber does not allow security borgs due to abuse, so repurposed it. I know it does not match weaponry 1-1 but it is not entirely unfitting, either.

File: modular_zubbers/code/modules/borgs/code/robot_defines.dm [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #define CYBORG_ICON_MED_WIDE_BUBBER 'modular_zubbers/code/modules/borgs/sprites/widerobot_med.dmi'
 #define CYBORG_ICON_MED_TALL_BUBBER 'modular_zubbers/code/modules/borgs/sprites/tallrobot_med.dmi'
 #define CYBORG_ICON_MED_LARGE_BUBBER 'modular_zubbers/code/modules/borgs/sprites/largerobot_med.dmi'
+#define CYBORG_ICON_MED_HAYDEE_BUBBER 'modular_zubbers/code/modules/borgs/sprites/boobrobot_med.dmi'
```
Comment (StrangeWeirdKitten) [inline modular_zubbers/code/modules/borgs/code/robot_defines.dm:11]:
```suggestion
#define CYBORG_ICON_MED_HAYDEE_BUBBER 'modular_zubbers/code/modules/borgs/sprites/boobrobot_med.dmi'
```
Can you rename all the dmi files to something else? Just call them Haydeez_med or whatever.

File: modular_zubbers/code/modules/borgs/code/robot_model.dm [hunk 2]
```diff
@@ -702,3 +746,13 @@
 #undef CYBORG_ICON_TYPE_SEC_CATBORG
 #undef CYBORG_ICON_TYPE_SYNDI_CATBORG
 #undef CYBORG_ICON_TYPE_NINJA_CATBORG
+
+//Haydeez borgs are nuts
```
Comment (StrangeWeirdKitten) [inline modular_zubbers/code/modules/borgs/code/robot_model.dm:750]:
```suggestion
```

**Review by StrangeWeirdKitten — CHANGES_REQUESTED** [review]

File: modular_zubbers/code/modules/borgs/code/robot_model.dm (hunk 2 above)
Comment (GyrPhlymBaBumble) [reply modular_zubbers/code/modules/borgs/code/robot_model.dm:750]: > 

Why? This is the same as line 736 and  723. Should we not remain consistent?

File: modular_zubbers/code/modules/borgs/code/robot_defines.dm (hunk 1 above)
Comment (GyrPhlymBaBumble) [reply modular_zubbers/code/modules/borgs/code/robot_defines.dm:11]: Why?

Comment (GyrPhlymBaBumble) [reply modular_zubbers/code/modules/borgs/code/robot_defines.dm]: > Can you rename all the dmi files to something else? Just call them Haydeez_med or whatever.

Done. Changed boob to curve.

**Comment by StrangeWeirdKitten** [comment]
Going to run a short TM period to ensure nothing is broken due to this PR being made entirely verified. Other then that, that's all of my issues with it resolved.

**Review by projectkepler-ru — COMMENTED** [review]
I'm personally against this now, since it look and feel extremely out of place

**Comment by StrangeWeirdKitten** [comment]
> HaydeeBunvice I did make a more clothed version of the one service bot and will be doing the same to the other, as they are the ones that are most problematic. I am wondering if there is a way to let the borgs have an alt-skin without using the dom module or however that works.
> 
> I don't have a DMI made yet and won't implement it yet.

The only thing that allows alt skins in the code is pretty much having them as an alt option on the selection wheel.

**Comment by GyrPhlymBaBumble** [comment]
> > HaydeeBunvice I did make a more clothed version of the one service bot and will be doing the same to the other, as they are the ones that are most problematic. I am wondering if there is a way to let the borgs have an alt-skin without using the dom module or however that works.
> > I don't have a DMI made yet and won't implement it yet.
> 
> The only thing that allows alt skins in the code is pretty much having them as an alt option on the selection wheel.

Really? The research borgs have shield sprites and if there is a way to activate them then that method could be used for alting.

**Comment by pixelkitty286** [comment]
> > > HaydeeBunvice I did make a more clothed version of the one service bot and will be doing the same to the other, as they are the ones that are most problematic. I am wondering if there is a way to let the borgs have an alt-skin without using the dom module or however that works.
> > > I don't have a DMI made yet and won't implement it yet.
> > 
> > 
> > The only thing that allows alt skins in the code is pretty much having them as an alt option on the selection wheel.
> 
> Really? The research borgs have shield sprites and if there is a way to activate them then that method could be used for alting.

This will probably be removed soon since the code for shields is to involved with my component system

**Comment by Swiftfeather** [comment]
I don't think these should be merged on bubberstation.

**Comment by Shadow-Quill** [comment]
These feel pretty out of place, especially the service module.

**Comment by ReturnToZender** [comment]
These would be great as synth sprites, but they look too much like a synth to fit borgs. Borgs can be sexualized but with future, lawed synths I feel these sprites would be superb. They look just like the dollhouse set and I think would fit in very nicely.

**Comment by GyrPhlymBaBumble** [comment]
image

**Pull request merged**